pkg/agent/manager/cache/cache.go

```diff
@@ -113,22 +136,40 @@ type Cache struct {
 	// selectors holds the selector indices, keyed by a selector key
 	selectors map[selector]*selectorIndex
 
-	// staleEntries holds stale registration entries
+	// staleEntries holds stale or new registration entries which require new SVID to be stored in cache
 	staleEntries map[string]bool
 
 	// bundles holds the trust bundles, keyed by trust domain id (i.e. "spiffe://domain.test")
 	bundles map[spiffeid.TrustDomain]*bundleutil.Bundle
+
+	// svids are stored by entry IDs
+	svids map[string]*X509SVID
+
+	// maxSVIDCacheSize is a soft limit of max number of SVIDs that would be stored in cache
+	maxSvidCacheSize int
+
+	// svidCacheExpiryPeriod is a period after which svids that don't have subscribers will be removed from cache
+	svidCacheExpiryPeriod time.Duration
 }
 
-// StaleEntry holds stale entries with SVIDs expiration time
+// StaleEntry holds stale or outdated entries which require new SVID with old SVIDs expiration time (if present)
 type StaleEntry struct {
 	// Entry stale registration entry
 	Entry *common.RegistrationEntry
 	// SVIDs expiration time
 	ExpiresAt time.Time
 }
 
-func New(log logrus.FieldLogger, trustDomain spiffeid.TrustDomain, bundle *Bundle, metrics telemetry.Metrics) *Cache {
+func New(log logrus.FieldLogger, trustDomain spiffeid.TrustDomain, bundle *Bundle, metrics telemetry.Metrics,
+	maxSvidCacheSize int, svidCacheExpiryPeriod time.Duration, clk clock.Clock) *Cache {
+	if maxSvidCacheSize == 0 {
```

Comment:
```suggestion
	if maxSvidCacheSize > 0 {
```
and maybe logging / error on negative value?

same for `svidCacheExpiryPeriod`

Reply:
Added unit test here but ignoring bad values as we added validations when parsing config.